src/js/form.js

```diff
@@ -530,7 +530,8 @@ Form.prototype.getRelatedNodes = function( attr, filter, updated ) {
     filter = filter || '';
 
     // The collection of non-repeat inputs, calculations and groups is cached (unchangeable)
-    if ( !this.nonRepeats[ attr ] ) {
+    // We essentially exclude outputs from the cache, because outputs can be added via itemsets (in labels)
+    if ( !this.nonRepeats[ attr ] || filter === '.or-output' ) {
```

Comment:
The caching in getRelatedNodes causes an issue for outputs in dynamically update itemset labels. An itemset update doesn't trigger a cache refresh.

It's crude to completely disable caching for outputs though. An alternative would be to only bust the cache if an itemset update deals with options that have outputs.

Reply:
It's the second caching change that can be rolled back, that you discovered, I believe. Not this one.

The alternative implementation would create complexity. Either directly clearing the cache in itemset.js (which seems wrong), or use a new event and listener, I think.